Update PyTEAL docs about how to provide external parameters (#148)
scale-it/algorand-builder: docs/py-teal.md

# Usage

- `algob compile` - compiles all files in `assets` folder with `.teal` as well as `.py` extension

# External Parameters Support

- We can use parameters in the PyTEAL code, however that parameters are hardcoded.
If the address is loaded dynamically (eg from KMD), we can't use PyTEAL code prior to the address knowledge.
- If we have to deploy same contract multiple times with only difference of initialization variables, we will have to change the variables in PyTeal code everytime we deploy it.
- To solve this problem we have introduced a support for passing `external parameters`.
- Deployer functions(`loadLogic`, `fundLsig`, `deploySSC`) takes one extra optional argument: a smart contract initialization parameters object(`scInitParam`). This argument is used (when not empty) when running PyTEAL compilation.

> - Parameter objects are part of hash. So if we change parameters, we may need to redeploy the smart contract.

Reviewer: ```suggestion
- Changing parameters will change a generated TEAL code. Hence it the Delegated Signature or Smart Contract address will be different and we may need to redeploy it.
```

> To use this feature, you can pass an external parameter object from the script: 

Reviewer: ```suggestion
### Usage

To use this feature, you can pass an external parameter object in a deployment script: 
```
